php/libraries/Module.class.inc

```diff
@@ -351,78 +365,31 @@ abstract class Module extends \LORIS\Router\PrefixRouter
 
         $user  = $request->getAttribute("user") ?? new \LORIS\AnonymousUser();
         $loris = $request->getAttribute("loris");
-        try {
-            $page = $this->loadPage($loris, $pagename);
-            // FIXME: Hack required for breadcrumbs. This should be removed,
-            // but some tests depend on it.
-            if ($this->getName() === $pagename) {
-                $_REQUEST['test_name'] = $pagename;
-            } else {
-                $_REQUEST['subtest'] = $pagename;
-            }
 
-            $page->loadResources($user, $request);
-            if ($page->_hasAccess($user) !== true) {
-                return (new \LORIS\Middleware\PageDecorationMiddleware(
-                    $user
-                ))->process(
-                    $request,
-                    new \LORIS\Router\NoopResponder(
-                        new \LORIS\Http\Error(
-                            $request,
-                            403,
-                            "You do not have access to this page."
-                        )
-                    )
-                );
-            }
-        } catch (\NotFound $e) {
+        $page = $this->loadPage($loris, $pagename);
+        // FIXME: Hack required for breadcrumbs. This should be removed,
+        // but some tests depend on it.
+        if ($this->getName() === $pagename) {
+            $_REQUEST['test_name'] = $pagename;
+        } else {
+            $_REQUEST['subtest'] = $pagename;
+        }
+
+        $page->loadResources($user, $request);
+        if ($page->_hasAccess($user) !== true) {
             return (new \LORIS\Middleware\PageDecorationMiddleware(
                 $user
             ))->process(
                 $request,
                 new \LORIS\Router\NoopResponder(
                     new \LORIS\Http\Error(
                         $request,
-                        404,
-                        "File not found: " . $request->getURI()->__toString()
+                        403,
+                        "You do not have access to this page."
                     )
                 )
             );
-            /* The order of these catch statements matter and should go from
```

Comment:
This was moved to the ExceptionHandlingMiddleware (where it should have been in the first place) so that the exception log handler is used.